aggregator/src/aggregation/rlc/gates.rs

```diff
@@ -408,11 +423,18 @@ impl RlcConfig {
         offset: &mut usize,
     ) -> Result<AssignedCell<Fr, Fr>, Error> {
         // when a and b are both small (as in our use case)
-        // if a < b, (a-b) will under flow and the highest bit of (a-b) be one
-        // else,  the highest bit of (a-b) be zero
-        let sub = self.sub(region, a, b, offset)?;
-        let bits = self.decomposition(region, &sub, offset)?;
-        Ok(bits[253].clone())
+        // if a >= b, c = 2^32 + (a-b) will be >= 2^32
+        // we bit decompose c and check the 33-th bit
+        let two_to_thirty_two = self.load_private(region, &Fr::from(1 << 32), offset)?;
+        let two_to_thirty_two_cell =
+            self.two_to_thirty_two_cell(two_to_thirty_two.cell().region_index);
+        region.constrain_equal(two_to_thirty_two_cell, two_to_thirty_two.cell())?;
+
+        let ca = self.add(region, &two_to_thirty_two, a, offset)?;
+        let c = self.sub(region, &ca, b, offset)?;
+        let bits = self.decomposition(region, &c, offset)?;
```

Comment:
The invocation of `self.decomposition` still is buggy because it decompose a field element into 254 bits and the `Fr` can only support 253 bit decomposition safely. I think it will be better to add one more parameter `nBits` in the decomposition function to enforce that the input `c` must be within the range `[0, 2^{nBits})`. And we can then use 
```rust
let bits = self.decomposition(region, &c, 33, offfset)?;
```

Reply:
fixed in https://github.com/scroll-tech/zkevm-circuits/commit/00ba5ff671d5074a1f508be2582f8c5c664072d5

(fix pushed to #712 because it also used decompose)